libs/wingsdk/src/shared-aws/api.ts

```diff
@@ -68,3 +70,28 @@ export class Api {
     );
   }
 }
+
+/**
+ * A helper class for working with AWS api endpoint handlers.
+ */
+export class ApiEndpointHandler {
```

Comment:
I feel like expanding our API for this isn't needed when wing can already handle this in a more powerful way https://github.com/winglang/wing/issues/6334#issuecomment-2077788152

Enumerating all the ways we can map inflights to each other seems like an uphill battle.